### ISSUE TITLE
FY22 Backlog: Add TCP Review Fee to SP, GP

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#6300]
### A new TCP Review Fee was approved for FY22 which requires process updates to the SP and GP folders in AMANDA.
- _This request was put in the backlog until after the AMANDA V7 upgrade. It also requires extensive coordination with DSD._ 

**Stakeholders:**
- (ATD) Ryan, Darren, Laura
- (DSD) Deborah F, Zachary W., Roderick B, Zackary M.

### Summary
- **SP:** ATD Staff will identify the TCP Fee requirement during CC, and the fee will be included in the big invoice that intake creates after CC. 
- **GP:** When ATD staff identifies the TCP Fee requirement, they will use a folder process to insert the fee as a stand-alone invoice.

### SP folder Acceptance Criteria
**The following process info needs to be configured in the AMANDA Admin Module**
- Info Description: TCP Review Fee Required? 
- Info Type: Boolean
- Info Default Value: Yes
- Info Process: CC Traffic Control Process (51830)

**The following fee logic needs to be added for SP folder fees**
- Process Info field 'TCP Review Fee Required?' needs to be added to CC Traffic Control Process (51830)
- When user makes attempt 'Completed without Site Visit' on Completeness Check Distribution Process (51116) AND Info field 'TCP Review Fee Required?' Process Info in CC Traffic Control Process (51830) is answered 'Yes', System must insert TCP Review Fee and ATD Tech Surcharge Fee (50174).
    - The two fees should insert along with the other fees that are inserted w/ this procedure / They should not be on a stand-alone bill unless no other fees are inserted. 
- The system should not waiver the TCP Review Fee and ATD Tech Surcharge Fee when other SP fees are waived (Fee waiver logic exists in SMART Housing Folder Infos 50920 and 50921).

### GP Folder Acceptance Criteria
**The following Process attempt needs to be added in AMANDA Admin Module**
Attempt Description - Add TCP Fee
Process - Traffic Control Review (51300)

**The following fee logic needs to be added for GP folders**
- Attempt 'Add TCP Fee' needs to be added to Traffic Control Review Process (51300)
- When user makes attempt Add TCP Fee in Traffic Control Review Process, system should insert TCP Review Fee and ATD Tech Surcharge Fee. 
- System should also bill the inserted TCP Review Fee and ATD Tech Surcharge Fee as part of the same bill 
- The Add TCP Fee attempt should not close the Traffic Control Review Process 

----
**Running Task List**
- [x] (Tracy, 6/3/21) Write draft AC; send to stakeholders for review
- [x] (Tracy) Schedule time w/ Zach for GP training
- [x] (Tracy, 6/23) Write possible solutions for GP specifically
- [x] (Tracy, 6/23) Send draft AC for both SP & GP to Ryan & Darren when they're ready

**Comment by TracyLinder** [comment]
### Stakeholder Meeting 6/3/21

**Notes:**
Existing TCP related fields (on SP):
Traffic Control Review? (Y or N)


ATD will be responsible for triggering the fee on DSD folders. This would occur during the completeness check formal review. 

Can ATD do this work, or will DSD want to do this
DSD is open to LIT doing the work, so long as 
Can we use a process info on 51300 Traffic Control Review or Or on the 51830 CC Traffic Control?
Info Name: TCP Review Fee Required?
Alert message for users to be reminded to complete the process info. 
Auto-bill the fee when inserted
Ryan/Intake would prefer this fee be inserted along with the other misc. review fees, on one bill for all. Deborah is concerned this would cause problems when there are waivers on all fees but not TCP.
Invoice to the folder applicant
SP & GP will have a different process. 
The 4% surcharge fee applies.

Will the fee ever be waived for smart housing or school project? 

Who does this fee not apply to? How will the system know not to charge?

For GP the system is inserting the fee, then waiving to zero and paying a $0 invoice. 

GP has an info: Fee waiver requested?

**Comment by TracyLinder** [comment]
### Risks of billing TCP fee with existing review fees that have waivers:
_From Deborah F._
If SH is required to be waived 100% or a portion of the fees, need to validate that the other fees are receiving waivers where ATD fee will not receive the waiver. Test where possibly fees for DSD fees are waived & add ATD fees where it should not be waived. Possibly add DSD fees first to be waived, then ATD fees where they should not be waived, then add another DSD fee, this is to validate how the fees are inserting before & after ATD fees are added. Make sure to bill them & see if the math adds up correctly. Be sure to test Portal also to make sure the fees add up correctly.

**Comment by TracyLinder** [comment]
![image.png](https://images.zenhubusercontent.com/5cca1e522b8dfc0ee14dedf5/4f3a31ef-0b28-4fc8-8388-24e9a1050ece)

**Comment by TracyLinder** [comment]
![Screen Shot 2021-06-10 at 2.46.30 PM.png](https://images.zenhubusercontent.com/5cca1e522b8dfc0ee14dedf5/f3e2e1a9-29f1-4142-865e-fa45a510d419)

**Comment by TracyLinder** [comment]
I'm at a stoping point with discovery & need stakeholder feedback. 
I scheduled a meeting on 6/14 w/ Ryan, Darren, Jaclyn, and Isiah to continue working on this.

**Comment by TracyLinder** [comment]
**Meeting on 6/21 didn't go well; the right people were not invited.**
- Another mtg w/ the correct people is scheduled for 6/23
_Tracy, Darren, Laura, Zachary Witaker, Zackary Martin_

**Comment by TracyLinder** [comment]
**Email to Ryan & Darren**

Howdy,

Darren & I met with DSD intake staff today to flesh out some of the technical and workflow details regarding adding the TCP Review fee to SP & GP folders. 

The tech solution & workflow for SP are ready to review with all stakeholders. GP - not so much. I'll be trying to ingest what we learned today and propose possible solutions. 

Notes on what we learned about the GP are below. Ryan - I have 4 questions for you below as well. 

**Questions:**

1. Does the 4% tech surcharge apply to all TCP Review fees? (Even for city departments)
2. Departments who do pay GP permit fees are currently doing so on an invoice-by-invoice basis, using the JV process. Will they also use the JV process to pay the TCP review fee?
3. Will non-city entities ever be billed for a TCP review fee on the GP folder?
4. Who is currently performing the TCP review process on the GP folder? (Laura I'm assuming.)

**Notes on GP:**
The majority of GP folders are created as a child of the UC folder

The majority GP applicants are city departments

There is not a CC process for GP in AMANDA. CC is occurring outside of AMANDA using other software. 

CC happens before the GP folder is created, then the folder is manually created by DSD staff

Traffic Control Review? (61135) info doesn’t always default to NO; we’re not sure what the system uses to determine the value. 

If created manually from scratch, it defaults to yes, 

if created as a child of UC it defaults to No and then DSD intake staff manually updates it to yes. 

I’d like approval from DSD to update this to always defaulting to yes, regardless of how the folder is created. 

For most GP folders, the fees are waived. This is how that works:

During folder creation, intake staff uses folder info Fee Waivers Requested (75140) to identify if fees will be waived. 

If info = yes, after saving and moving to the fee tab, the system inserts the fees and adjusts them to $0. (Tracy, test this) (Does the system invoice at this time?)

If info = no, after saving and moving to the fee tab, the system inserts the fees but does not bill them, or adjust them

When fees are required they are manually billed by DSD around the time that an ‘entered’ attempt is saved on the General Permit Intake process.

City Departments generally are the ones paying GP folders fees, they always pay with a JV. 

DSD cashier then pays the invoice using the JV payment type. Then staff send the receipt to the paying department. 

Will a non-city department ever be billed the TCP review fee on a GP folder? - We don’t think so. 

ATD does not currently charge city departments the out of process review fee. This changes on 10/1 - everyone other than franchised utilities will have to pay this fee.



Thanks,


Tracy Linder, Product Manager

**Comment by TracyLinder** [comment]
I sent my GP notes to Zach & Zack, and asked them to let me know if anything is incorrect.

**Comment by TracyLinder** [comment]
### 7/8/21 w/ Laura & Vish

**Summary:** 
- **SP:** TCP Fee cannot be included on the multi-department review invoice. This will need to be charged on the GF folder. 
- **GP:** Very, very few of these permits include fees. My suggestion is that we use the GF folder instead. 

**Notes:**
- Over 70% of TCP reviewed are deferred 
- We don't know during SP Completeness check if there will be an in-process fee.
    - On TCP CC, Laura either uses 'Complete' or 'No Review Required' attempts
- Most experienced engineers know to add the TCP note on the CC submittal, but some don't include it. 
    - It's a long paragraph that includes if the TCP will be deferred or not
    - There could also be a sheet index that includes the TCP 

![Screen Shot 2021-07-08 at 3.30.00 PM.png](https://images.zenhubusercontent.com/5cca1e522b8dfc0ee14dedf5/fe389093-1e80-4d66-b942-ca158482e86a)

**Comment by TracyLinder** [comment]
### 7/14 Discussion
_Tracy, Ryan, Darren, Laura_

**SP**
- Default process info to Yes
- Laura can change it to No when applicable


**GP**
- Laura will charge TCP fee manually on the fee tab. 

**Action Items**
(Tracy) Send DSD email w/ proposed solutions. 
Deborah, Elizabeth Boswell, John McDonald, Donna Gollati, Maria Jaramillo, both Zachs.

**Comment by TracyLinder** [comment]
### 8/4 Sync - Tracy & Vish

- On SP, we need to know more about how the system knows to apply adjustments/waivers to the review fees. We need to exclude the TCP Review Fee from this code. 

![Screen Shot 2021-08-04 at 11.43.58 AM.png](https://images.zenhubusercontent.com/5cca1e522b8dfc0ee14dedf5/5acf4a23-00ee-48fc-8f2f-cddf92774385)

**Comment by TracyLinder** [comment]
**Update**
- Vish to work on AC
- We will sync w/ Sree & Nate on Friday to review AC

**Comment by TracyLinder** [comment]
This will be pulled in after #6844

**Comment by Sreelatha-BlazeKannan** [comment]
@Nadin-Nader , @TracyLinder - Working on this Card. Analysis is in Progress

**Comment by TracyLinder** [comment]
### Update

Darren asked where this is in priority. He let us know this is causing frustration with DSD and the customer because the fee is becoming a bottleneck.

**Comment by TracyLinder** [comment]
### Currently Darren's # 1 AMANDA priority. 
- Next step: Review & Update AC. Tracy will sync w/ Vish as a start.

**Comment by TracyLinder** [comment]
**Update/Note:** I want to review this again with DSD. I'll bring Darren into that meeting too.

**Comment by TracyLinder** [comment]
### Email to Zach W. with DSD

Hi Zachary,

You probably remember conversations with me in the summer of 2021 about adding a traffic control plan review fee to SP and GP folders in AMANDA. Copied is the supervisor over TCP reviews, Darren.

We're now ready to move forward with adding the fee so I'd like to meet with you to review the [acceptance criteria we have documented](https://github.com/cityofaustin/atd-data-tech/issues/6300). I'm thinking it would be most efficient if the 3 of us meet initially to sync, then we can bring in any other stakeholders who should be part of the conversation. Let me know your thoughts; I'll wait for your response before I schedule anything. 

Thanks, 

Tracy Linder, Product Manager

**Comment by TracyLinder** [comment]
@vschandramouli I just realized we'll have to update the AC to include Cost of Service considerations.

**Comment by vschandramouli** [comment]
Meeting with stakeholders scheduled for 1/30/23 to review and finalize AC.

**Comment by TracyLinder** [comment]
## AC Review w/ DSD & ROW


## SP
- [ ] Confirm w/ Ryan & Darren that smart housing doesn't wave the TCP Review fee. (Do email, copy Zach, Kenneth). 
- [ ] Zach is concerned the ATD Tech Surcharge could be very messy. So lets keep an eye on that during UAT. Tracy to speak to the developer about this before work starts. 

- When zach saves 'Complete w/o site visit' the fees should insert but not charge. This is already in the AC, just making another note. 

## GP
- How does the invoice get to the customer? 
    - Add to the review comment something like "An invoice has been added to your permit, please go online and pay it."
    - The customer can ask DSD if they need help at that point.

**Comment by TracyLinder** [comment]
**Hello Ryan and Darren,**

I met with Zach and Kenneth with DSD this morning to get their feedback on our plan to add the TCP Review fee to the SP and GP folders. DSD would like confirmation from you that SMART housing waivers do not apply to the TCP Review fee. I offered to start an email thread on the topic. Please advise. 

Thank You,

Tracy Linder, Product Manager

**Comment by TracyLinder** [comment]
### Sync w/ Ryan & Darren

**Summary:** It's likely we don't need to do this work anymore. Their current fee assessment process is working well for them. Darren will talk to Alyse about it and get back to me. FYI @vschandramouli 

- 95% of SP are deferring and the customer pays through the TCP intake process. Darren has only seen one example in the last 2 years of a TCP being reviewed as part of the SP. 
- It's easier for TCP Intake to assess fees during the TCP Intake process.

**Comment by TracyLinder** [comment]
Sent Darren a Teams chat asking for an update.

**Comment by TracyLinder** [comment]
### Update & Closing
- Darren confirmed all TCP fees will be charged on the GF folder. We can close this request now.
